server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -2320,11 +2320,14 @@ protected List<AuxTransport> newAuxTransports(NetworkModule networkModule) {
         return networkModule.getAuxServerTransportList();
     }
 
+    /**
+     * Base factory for creating DiscoveryNode instances during node initialization.
+     */
     private static class LocalNodeFactory implements Function<BoundTransportAddress, DiscoveryNode> {
         private final SetOnce<DiscoveryNode> localNode = new SetOnce<>();
         private final String persistentNodeId;
-        private final Settings settings;
-        private final RemoteStoreNodeService remoteStoreNodeService;
+        protected final Settings settings;
+        protected final RemoteStoreNodeService remoteStoreNodeService;
```

Comment:
Nitpick, but this field can move to `RemoteStoreVerifyingLocalNodeFactory` since it's not used here, right?